src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -1049,31 +1049,43 @@ private void createFlattenedDependenciesDirect( List<Dependency> projectDependen
      * @throws ArtifactDescriptorException
      */
     private void createFlattenedDependenciesAll( List<Dependency> projectDependencies, List<Dependency> flattenedDependencies )
-            throws DependencyTreeBuilderException, ArtifactDescriptorException {
-        final Queue<DependencyNode> dependencyNodeLinkedList = new LinkedList<DependencyNode>() {};
+            throws DependencyTreeBuilderException, ArtifactDescriptorException
+    {
+        final Queue<DependencyNode> dependencyNodeLinkedList = new LinkedList<DependencyNode>();
         final Set<String> processedDependencies = new HashSet<>();
 
         final Artifact projectArtifact = this.project.getArtifact();
 
         final DependencyNode dependencyNode = this.dependencyTreeBuilder.buildDependencyTree(this.project,
                 this.localRepository, null);
 
-        dependencyNode.accept(new DependencyNodeVisitor() {
-            @Override public boolean visit(DependencyNode node) {
-                if (node.getArtifact().getGroupId().equals(projectArtifact.getGroupId()) && node.getArtifact().getArtifactId().equals(projectArtifact.getArtifactId())) {
+        dependencyNode.accept(new DependencyNodeVisitor()
+        {
+            @Override public boolean visit(DependencyNode node)
+            {
+                if (node.getArtifact().getGroupId().equals(projectArtifact.getGroupId()) && node.getArtifact().getArtifactId().equals(projectArtifact.getArtifactId()))
+                {
                     return true;
                 }
-                if (node.getState() != DependencyNode.INCLUDED) return true;
+                if (node.getState() != DependencyNode.INCLUDED) {
+                    return true;
```

Comment:
i know this is in the previous code, but should this actually return false, as we discussed in a separate call in the past?

Reply:
I think that returning true will omit the current depNode but still visit its children. I will check whether it will make any difference if we return false.

Reply:
quick ping ot see if we've checked this?

Reply:
@saturnism all tests passed if it returns false.